relayer/processor/path_processor.go

```diff
@@ -716,32 +649,34 @@ func (pp *PathProcessor) processLatestMessages(ctx context.Context) error {
 	wg.Wait()
 
 	// concatenate applicable messages for pathend
-	var pathEnd1Messages, pathEnd2Messages []provider.RelayerMessage
+	var pathEnd1PacketMessages, pathEnd2PacketMessages []packetIBCMessage
+	var pathEnd1ConnectionMessages, pathEnd2ConnectionMessages []connectionIBCMessage
+	var pathEnd1ChannelMessages, pathEnd2ChannelMessages []channelIBCMessage
```

Comment:
It looks like these could all be pre-sized properly. OTOH since there are only two appends, it might only get resized once. I'm not sure what the current compiler does in this case and I haven't benchmarked it.

Reply:
good call, I moved the assembly of these slices out into their own functions with pre-sizing.